src/main/java/billgates/usecases/user_join/UserJoinUseCase.java

```diff
@@ -0,0 +1,29 @@
+package billgates.usecases.user_join;
+
+import billgates.database.QueryUserData;
+import billgates.interface_adapters.DatabaseGateway;
+
+public class UserJoinUseCase implements UserJoinInputPort {
+    private final DatabaseGateway gateway;
+    private final UserJoinOutputPort outputPort;
+
+    public UserJoinUseCase(DatabaseGateway gateway, UserJoinOutputPort outputPort){
+        this.gateway = gateway;
+        this.outputPort = outputPort;
+    }
+
+    @Override
+    public void join(UserJoinRequestModel model) {
+        QueryUserData userData = gateway.getUserData();
+        if (!userData.getUsers().contains(model.getUsername())){
+            // TODO add the user to the database
+
+            // TODO log the user in
+
+        }
+        else {
+            // pop a dialog and notify the user that the name is invalid
```

Comment:
What do you mean by 

> the name is invalid

?
In this else branch, if this user exists in our database, we should then check whether the input password matches with the password in our database
1. If the passwords don't match, then we should pop a dialog and notify the user that 
   1) maybe they type in an incorrect password OR 
   2) Since the user already exists in our database, we cannot register a new user with the same username

2. If the passwords match, we should then login